db/attachment.go

```diff
@@ -180,27 +190,35 @@ func (db *Database) loadAttachmentsData(attachments AttachmentsMeta, minRevpos i
 	return newAttachments, nil
 }
 
-// Retrieves an attachment given its key.
-func (db *Database) GetAttachment(key AttachmentKey) ([]byte, error) {
-	v, _, err := db.Bucket.GetRaw(attachmentKeyToString(key))
+// DeleteAttachmentVersion removes attachment versions from the AttachmentsMeta map specified.
+func DeleteAttachmentVersion(attachments AttachmentsMeta) (newAttachments AttachmentsMeta) {
+	newAttachments = attachments.ShallowCopy()
```

Comment:
Why do we need to make a shallow copy here?  It looks like it's being called based on the rev cache value, which is already a shallow copy.